model/perf.go

```diff
@@ -656,13 +657,90 @@ func (r *PerformanceResults) FindOutdatedRollups(ctx context.Context, name strin
 	return errors.WithStack(it.Close(ctx))
 }
 
-func GetTimeSeriesIds(ctx context.Context, env cedar.Environment) ([]TimeSeriesId, error) {
-	pipeline := []bson.M{
+func createTimeStamp(ctx context.Context, env cedar.Environment, id TimeSeriesId, field string) error {
+	_, err := env.GetDB().Collection(metricMetadataCollection).UpdateOne(ctx, id, bson.M{
+		"$currentDate": bson.M{
+			field: true,
+		},
+	})
+	if err != nil {
+		return errors.Wrapf(err, "Unable to create timestamp")
+	}
+	return nil
+}
+
+func MarkMetricAsUpdated(ctx context.Context, env cedar.Environment, id TimeSeriesId) error {
+	err := createTimeStamp(ctx, env, id, "lastTimeSeriesUpdate")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to mark metric as updated")
+	}
+	return nil
+}
+
+func MarkMetricAsAnalyzed(ctx context.Context, env cedar.Environment, id TimeSeriesId) error {
+	err := createTimeStamp(ctx, env, id, "lastChangePointDetection")
+	if err != nil {
+		return errors.Wrapf(err, "Unable to mark metric as analyzed for change points")
+	}
+	return nil
+}
+
+func GetMetricsNeedingChangePointDetection(ctx context.Context, env cedar.Environment) ([]TimeSeriesId, error) {
+	cur, err := env.GetDB().Collection(metricMetadataCollection).Find(ctx, bson.M{
+		"$expr": bson.M{
+			"$lt": []string{
+				"$lastChangePointDetection",
+				"$lastTimeSeriesUpdate",
+			},
+		},
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to get metrics needing change point detection")
+	}
+	defer cur.Close(ctx)
+	var res []struct {
+		Id                       TimeSeriesId `bson:"_id"`
+		LastChangePointDetection time.Time    `bson:"lastChangePointDetection"`
+		LastTimeSeriesUpdate     time.Time    `bson:"lastTimeSeriesUpdate"`
+	}
+	err = cur.All(ctx, &res)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to decode metrics needing change results")
+	}
+	result := []TimeSeriesId{}
+	for _, val := range res {
+		result = append(result, val.Id)
+	}
```

Comment:
I'd just do an aggregation/projection to reshape the documents on the db rather than copying them out.